src/ILCompiler.CppCodeGen/src/CppCodeGen/ILToCppImporter.cs

```diff
@@ -917,6 +917,16 @@ private void ImportCall(ILOpcode opcode, int token)
                     return;
             }
 
+            if (method.IsRawPInvoke())
+            {
+                AppendLine();
+                Append("PInvokeTransitionFrame __piframe");
```

Comment:
This assumes that there will be at most one RawPInvoke callsite per method. It is a safe assumption to make today, but it maybe worth a comment.